hi-ml-histopathology/src/histopathology/utils/tiles_selection_utils.py

```diff
@@ -165,28 +165,29 @@ def update_slides_selection(self, batch: Dict[SlideOrTileKey, Any], results: Dic
             for i in range(batch_size):
                 label = results[ResultsKey.TRUE_LABEL][i].item()
                 probs_gt_label = results[ResultsKey.CLASS_PROBS][:, label]
-                self._update_label_slides(
-                    class_slides_heap=self.top_slides_heaps[label],
-                    tiles=batch[SlideKey.IMAGE][i],
-                    attn_scores=results[ResultsKey.BAG_ATTN][i].squeeze(),
-                    slide_node=SlideNode(
-                        slide_id=slide_ids[i],
-                        prob_score=probs_gt_label[i].item(),
-                        true_label=label,
-                        pred_label=results[ResultsKey.PRED_LABEL][i].item(),
-                    ),
-                )
-                self._update_label_slides(
-                    class_slides_heap=self.bottom_slides_heaps[label],
-                    tiles=batch[SlideKey.IMAGE][i],
-                    attn_scores=results[ResultsKey.BAG_ATTN][i].squeeze(),
-                    slide_node=SlideNode(
-                        slide_id=slide_ids[i],
-                        prob_score=-probs_gt_label[i].item(),
-                        true_label=label,
-                        pred_label=results[ResultsKey.PRED_LABEL][i].item(),
-                    ),
-                )
+                if results[ResultsKey.BAG_ATTN][i].numel() != 0:  # some slides have no tiles and therefore no attn
+                    self._update_label_slides(
+                        class_slides_heap=self.top_slides_heaps[label],
+                        tiles=batch[SlideKey.IMAGE][i],
+                        attn_scores=results[ResultsKey.BAG_ATTN][i].squeeze(),
+                        slide_node=SlideNode(
+                            slide_id=slide_ids[i],
+                            prob_score=probs_gt_label[i].item(),
+                            true_label=label,
+                            pred_label=results[ResultsKey.PRED_LABEL][i].item(),
+                        ),
+                    )
+                    self._update_label_slides(
+                        class_slides_heap=self.bottom_slides_heaps[label],
+                        tiles=batch[SlideKey.IMAGE][i],
```

Comment:
There is some repeated indexing here - you could store these values as variables to avoid indexing twice